base/stream.jl

```diff
@@ -886,21 +893,45 @@ function stop_reading(stream::LibuvStream)
     if stream.status == StatusActive
         ret = ccall(:uv_read_stop, Cint, (Ptr{Void},), stream)
         stream.status = StatusOpen
-        ret
+        return ret
     elseif stream.status == StatusOpen
-        Int32(0)
+        return Int32(0)
     else
-        Int32(-1)
+        return Int32(-1)
     end
 end
 
-function read!(s::LibuvStream, b::Vector{UInt8})
-    nb = length(b)
-    r = readbytes!(s, b, nb)
-    if r < nb
-        throw(EOFError())
+readbytes!(s::LibuvStream, a::Vector{UInt8}, nb = length(a)) = readbytes!(s, a, Int(nb))
+function readbytes!(s::LibuvStream, a::Vector{UInt8}, nb::Int)
```

Comment:
yeah, i'm hoping to address this function in some future PR. right now, every implementation of it has slightly different behavior, but in significant ways.